temporal_cluster_matching/DataInterface.py

```diff
@@ -303,4 +311,94 @@ def data_stack_to_rgb(self, images):
             image = image[:,:,[2,1,0]]
             image = utils.scale(1.1*image, 0, 2500)
             rgb_images.append(image)
-        return rgb_images
\ No newline at end of file
+        return rgb_images
+
+
+################################################################
+################################################################
+
+
+class PlanetaryComputerS2DataLoader(AbstractDataLoader):
+
+    def __init__(self, geoms, pc_subscription_key, search_start="2015-01-01", search_end="2019-12-31"):
+        pc.settings.set_subscription_key(pc_subscription_key)
+        self.geoms = geoms
+        self.time_range = f"{search_start}/{search_end}"
+
+    @lru_cache(maxsize=None)
```

Comment:
Is lru effect on performance big?

Reply:
Yes -- you don't want to keep having to call the Planetary Computer API if you already know the results.